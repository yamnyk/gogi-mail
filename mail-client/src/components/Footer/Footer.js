import React from 'react';
import {connect, useSelector} from "react-redux";

export const Footer = ({user, isOpenedUserCopy=true, toggleUserCopy}) => {
  // const user = useSelector((store) => (store.user));

  const handleToggle = () => {
    toggleUserCopy();
  };

  return (
    <div data-testid={"footer-container"} className="container" style={{fontSize: '10px', textAlign: 'right'}}>
      hush...all rights have been stolen from other developers, but they don't know anything about it
      <button data-testid="toggler" onClick={handleToggle}>Toggle user info</button>
      {
        isOpenedUserCopy &&
        <p>the person who stoled the data is - <span data-testid="person-copy">{user ? user.name : 'You'}</span></p>
      }
    </div>
  );
};

export default connect((store) => ({user: store.user}))(Footer);