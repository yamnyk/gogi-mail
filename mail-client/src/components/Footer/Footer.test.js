import React from "react";
import { render } from "@testing-library/react";
import userEvent  from "@testing-library/user-event";
import { Footer } from "./Footer";

const {click} = userEvent;

describe("Unit testing Footer", () => {
  test("smoke test Footer", () => {
    render(<Footer />);
  });
  test("footer has correct text when user is NOT logged in", () => {
    const { getByTestId } = render(<Footer />);
    const personCopy = getByTestId("person-copy");

    expect(personCopy.textContent).toBe("You");
  });
  test("footer has correct text when user is logged in", () => {
    const { getByTestId } = render(<Footer user={{ name: "Gogi" }} />);
    const personCopy = getByTestId("person-copy");

    expect(personCopy.textContent).toBe("Gogi");
  });
  test("footer container has inline styles", () => {
    const { getByTestId } = render(<Footer />);
    const footerContainer = getByTestId("footer-container");

    expect(footerContainer).toHaveStyle("font-size: 10px; text-align: right;");
  });
  test('Footer toggling user info works', () => {
    const toggler = jest.fn();
    const {getByTestId} = render(<Footer toggleUserCopy={toggler}/>);

    expect(toggler).not.toBeCalled();

    click(
      getByTestId('toggler')
    );

    expect(toggler).toBeCalled();
  });
});
