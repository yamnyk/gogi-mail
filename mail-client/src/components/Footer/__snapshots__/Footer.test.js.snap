// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing Footer footer container has inline styles 1`] = `
<div
  class="container"
  data-testid="footer-container"
  style="font-size: 10px; text-align: right;"
>
  hush...all rights have been stolen from other developers, but they don't know anything about it
  <p>
    the person who stoled the data is - 
    <span
      data-testid="person-copy"
    >
      You
    </span>
  </p>
</div>
`;
